x/feegrant/keeper/keeper.go

```diff
@@ -228,3 +288,30 @@ func (k Keeper) ExportGenesis(ctx sdk.Context) (*feegrant.GenesisState, error) {
 		Allowances: grants,
 	}, err
 }
+
+func (k Keeper) removeFromGrantQueue(ctx sdk.Context, exp *time.Time, allowanceKey []byte) {
+	key := feegrant.FeeAllowancePrefixQueue(exp, allowanceKey)
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(key)
+}
+
+func (k Keeper) addToFeeAllowanceQueue(ctx sdk.Context, grantKey []byte, exp *time.Time) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(feegrant.FeeAllowancePrefixQueue(exp, grantKey), []byte{})
+}
+
+// RemoveExpiredAllowances iterates grantsByExpiryQueue and deletes the expired grants.
+func (k Keeper) RemoveExpiredAllowances(ctx sdk.Context) {
+	exp := ctx.BlockTime()
+	store := ctx.KVStore(k.storeKey)
+	iterator := store.Iterator(feegrant.FeeAllowanceQueueKeyPrefix, sdk.InclusiveEndBytes(feegrant.AllowanceByExpTimeKey(&exp)))
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		store.Delete(iterator.Key())
+		expLen := len(sdk.FormatTimeBytes(time.Now()))
```

Comment:
We should use the sdk.Context block time here to avoid non-determinism

Reply:
This is just to get the length of the output of  `FormatTimeBytes(time)`, anyhow updated it, thanks!.